### PR TITLE
Documentation for Gateway API implementation

### PR DESCRIPTION
**Pull request by swathinsankaran**
This PR contains the documentation for the Gateway API implementation.

### CODE REVIEW

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+For the Tech preview, AKO imposes a restriction on the order of GatewayAPI object creation. An object that is referenced must be created first. For example, GatewayClass must be created before Gateway and Gateway before HTTPRoute creation. This restriction is only applicable to the Gateway API objects and will be removed in the future releases.
+
+#### High Availability Support
+
+For the Tech Preview, High Availability is not supported.
+
+#### Configuring Static IP address
+
+The AKO supports gateway objects with a single IPv4 address. The user can their preferred static IPv4 address by specifying `spec.addresses` in the Gateway object. A sample configuration is shown below:
```
Comment (pkoshtavmware) [inline docs/gateway-api/gateway-api-v1beta1.md:238]: - "The user can their" should be changed to "The user can specify their".
-  Might want to capitalise first letter in 'gateway' (proper noun)

Also , we haven't mentioned anything about child vs, Pool and PoolGroup naming conventions anywhere. Should we include that too ?

**Review by pkoshtavmware — CHANGES_REQUESTED** [review]

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    controllerName: "ako.vmware.com/avi-lb"
+  ```
+
+It is important that the `.spec.controllerName` value specified MUST match `ako.vmware.com/avi-lb` for AKO to honor the GatewayClass and corresponding Gateway API objects.
+
+**NOTE:** A GatewayClass named `avi-lb` will get installed as part of the helm install or upgrade when the user enables the Gateway API feature. 
+
+#### Gateway
+
+The Gateway object represents an instance of a service-traffic handling infrastructure by binding Listeners to a set of IP addresses. The AKO validates the Gateway object and updates the status as `Accepted`. Then, the AKO translates the Gateway and its configuration as a Parent VS. The listeners in Gateway will be translated as listeners in the Parent VS and secrets will be configured in the AVI controller and will be referenced in the same Parent VS. AKO updates the status of Gateway as `programmed` along with the VIP of the Parent VS once the VS creation is completed.
```
[the remark this answers is not in the document]
Comment (swathinsankaran) [reply docs/gateway-api/gateway-api-v1beta1.md:54]: I think the `Services` will be confusing to the user who is from a K8s background and new to controller wordings.

File: docs/gateway-api/gateway-api-v1beta1.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    addresses:
+    - type: IPAddress
+      value: 10.1.1.10
+  ```
+
+**NOTE:** AKO claims support for a single address of type IPAddress. The Gateway must be re-created to update the address.
+
+#### HTTPRoute
+
+The HTTPRoute object provides a way to route HTTP requests. The AKO models a child VS based on this object. Currently, AKO supports match requests based on the hostname, path, and header specified. The filters to specify additional processing of the requests will be added as policy in the child VS by the AKO. The filters of type `RequestHeaderModifier`, `RequestRedirect` and `ResponseHeaderModifier` are supported in the current release.
```
[the remark this answers is not in the document]
Comment (swathinsankaran) [reply docs/gateway-api/gateway-api-v1beta1.md:98]: yes, I have mentioned it below. vh_match will be new to the user, so I have mentioned match criteria instead.

File: docs/gateway-api/gateway-api-v1beta1.md (hunk 1 above)
Comment (swathinsankaran) [reply docs/gateway-api/gateway-api-v1beta1.md:238]: The pool, pg, and all are added above while explaining the httproute to child vs modeling. Pool and PG naming is one thing I would like to add. but that will change in the future.

**Comment by swathinsankaran** [comment]
@pkoshtavmware I have addressed the comments. Could you please re-review?

File: docs/gateway-api/gateway-api-v1beta1.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pkoshtavmware) [reply docs/gateway-api/gateway-api-v1beta1.md:98]: Okay

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Pull request merged**